File: C Programming/Assignments/Assignment 4/Q4.c
void main(){
    int r;
    printf("Enter the range from 1 to: ");
    scanf("%d",&r);
    for(int i=1; i<=r; i++){
        int no = i;
        int sum = 0;
        while(no > 0){
            int digit = no % 10;
            int fact = 1;
            for(int j=1; j<=digit; j++){
                fact = fact * j;
            }
            sum = sum + fact;
            no = no / 10;
        }
        if(sum == i){
            printf("%d ", i);
        }
    }
}

File: Java/C TO JAVA/Assignment 4/Q4.java
class Q4
{
public static void main(String args[]){
    int r;
    System.out.printf("Enter the range from 1 to: ");
    scanf("%d",&r);
    for(int i=1; i<=r; i++){
        int sum = 0;
        for(int j=1; j<i; j++){
            if(i % j == 0){
             sum = sum + j;
            }
        }
    if(sum == i){
        Systen.out.printf("%d ", i);
        }
    }
}

}